### PR TITLE
Add kwok release job

### PR DESCRIPTION
**Pull request by wzshiming**
kubernetes-sigs/kwok#18

### CODE REVIEW

**Comment by wzshiming** [comment]
Hi @michelle192837 , @listx

Do you have time to look at it?
Could you help me check, what else do I need to do for uploading image and binary?

``` console
=== FAIL: prow/cmd/webhook-server TestUpdateSecret/noDnsNames (0.00s)
[548](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/27128/pull-test-infra-unit-test/1559058400480858112#1:build-log.txt%3A548)
    helpers_test.go:145: secret was not updated
[549](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/27128/pull-test-infra-unit-test/1559058400480858112#1:build-log.txt%3A549)
    --- FAIL: TestUpdateSecret/noDnsNames (0.00s)
```

File: config/jobs/image-pushing/k8s-staging-kwok.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  # this is the github repo we'll build from; this block needs to be repeated for each repo.
+  kubernetes-sigs/kwok:
+    - name: post-kwok-push-images
+      cluster: k8s-infra-prow-build-trusted
+      annotations:
+        # this is the name of some testgrid dashboard to report to.
+        testgrid-dashboards: sig-scheduling, sig-k8s-infra-gcb
+      decorate: true
+      branches:
+        # Build tags also
```
Comment (Huang-Wei) [inline config/jobs/image-pushing/k8s-staging-kwok.yaml:11]: I noticed other sig-scheduling owned configs ([1](https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/k8s-staging-scheduler-plugins.yaml), [2](https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/k8s-staging-kueue.yaml)) carry the following lines:

```
        - ^master$
        - ^release-.*
```

File: config/jobs/image-pushing/k8s-staging-kwok.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        containers:
+          - image: gcr.io/k8s-staging-test-infra/image-builder:v20220428-ae431ed1aa
+            command:
+              - /run.sh
+            args:
+              # this is the project GCB will run in, which is the same as the GCR images are pushed to.
+              - --project=k8s-staging-kwok
+              - --scratch-bucket=gs://k8s-staging-kwok-gcb
+              - --env-passthrough=PULL_BASE_REF
+              - --with-git-dir
```
Comment (Huang-Wei) [inline config/jobs/image-pushing/k8s-staging-kwok.yaml:25]: Is this parameter mandatory?

Comment (wzshiming) [reply config/jobs/image-pushing/k8s-staging-kwok.yaml:25]: That `--with-git-dir` and `--env-passthrough` are not mandatory

`--with-git-dir`
https://github.com/kubernetes/test-infra/blob/1576e68c95ec61ed19a1f3e8934cf5b990c39d37/images/builder/main.go#L140-L142
`--env-passthrough` 
https://github.com/kubernetes/test-infra/blob/1576e68c95ec61ed19a1f3e8934cf5b990c39d37/images/builder/main.go#L158-L168
`--scratch-bucket`
https://github.com/kubernetes/test-infra/blob/1576e68c95ec61ed19a1f3e8934cf5b990c39d37/images/builder/main.go#L188-L191
`--project`
https://github.com/kubernetes/test-infra/blob/1576e68c95ec61ed19a1f3e8934cf5b990c39d37/images/builder/main.go#L184-L186

File: config/jobs/image-pushing/k8s-staging-kwok.yaml (hunk 1 above)
Comment (wzshiming) [reply config/jobs/image-pushing/k8s-staging-kwok.yaml:13]: No release and master, I was just trying to be lazy, now I'm making up for it
https://github.com/kubernetes-sigs/kwok/pull/18/commits/f5b23506db3c15055abbec9f22cfa99756d0c9ab

**Comment by Huang-Wei** [comment]
/lgtm

This looks quite similar when other sub-project raised the initial request.

@spiffxp not sure you have cycles to take a look and approve. Much appreciated!

**Comment by wzshiming** [comment]
> BucketNotFoundException: 404 gs://k8s-staging-kwok-gcb bucket does not exist.

https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-kwok-push-images/1560514872045735936

@listx
Could you have time to take a look?
I want to create `k8s-staging-kwok-gcb` bucket, how should I do it?

**Comment by listx** [comment]
It's been a while but IIRC `k8s-staging-*` GCS buckets still appear to be created here https://github.com/kubernetes/k8s.io/blob/79c673e878f6c31757dd726cdf5e5aa27ee9b786/infra/gcp/bash/ensure-staging-storage.sh#L142. Maybe open up an issue there or ask in slack (channel #sig-k8s-infra)?

**Comment by wzshiming** [comment]
Need to be create here https://github.com/kubernetes/k8s.io/pull/4129

**Pull request merged**